presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -115,7 +117,8 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
                 hiveSplit.getFileSize(),
                 hiveSplit.getFileModifiedTime(),
                 hiveSplit.getSchema(),
-                hiveTable.getCompactEffectivePredicate().intersect(dynamicFilter.transform(HiveColumnHandle.class::cast)),
+                hiveTable.getCompactEffectivePredicate(),
+                dynamicFilter.transform(HiveColumnHandle.class::cast),
```

Comment:
why this split is needed? IMO intersected domain will be even smaller than `dynamicFilter` on its own

Reply:
I think we only need to check partition keys against `dynamicFilter`. Checking them against `effectivePredicate` may not be necessary.

Consider a case when `dynamicFilter` is `TupleDomain.all()` whereas `effectivePredicate` isn't. By separating them, the checking work is totally avoided (because `dynamicFilter` is `TupleDomain.all()`)